### PR TITLE
feat: lock terraform version to 1.4.6

### PR DESCRIPTION
**Pull request by sanketnadkarni**
Changes as below:
- added steps to use `tfenv` to the README.md
- added .terraform-version which acts as a configuration file for `tfenv`
- added `required_version` in provider.tf to use terraform version 1.4.6 ; reference: https://developer.hashicorp.com/terraform/tutorials/configuration-language/versions#review-example-configuration
- update setup_keys.sh script 

To Do: 
- currently all the above changes are only made to the dev folder, same needs to be replicated to the other envs.

```bash
❯ terraform version
Terraform v1.7.0
on darwin_amd64
+ provider registry.terraform.io/cyrilgdn/postgresql v1.11.0
+ provider registry.terraform.io/hashicorp/random v3.6.0

Your version of Terraform is out of date! The latest version
is 1.7.4. You can update by downloading from https://www.terraform.io/downloads.html


# If incorrect version is used terraform init command will fail with below error: 
❯ terraform init -backend-config=backend-config.tfvars

Initializing the backend...
Initializing modules...
╷
│ Error: Unsupported Terraform Core version
│
│   on provider.tf line 11, in terraform:
│   11:   required_version = "~> 1.4.6"
│
│ This configuration does not support Terraform version 1.7.0. To proceed, either choose another supported Terraform version or update this version constraint. Version constraints are
│ normally set for good reason, so updating the constraint may lead to other errors or unexpected behavior.
╵




# .terraform-version file will not allow to use incorrect version as seen below
❯ tfenv use 1.7.0
Switching default version to v1.7.0
Default version file overridden by /gs/treetracker-infrastructure/database-grants/terraform/dev/.terraform-version, changing the default version has no effect
Default version (when not overridden by .terraform-version or TFENV_TERRAFORM_VERSION) is now: 1.7.0




```

### CODE REVIEW

File: database-grants/terraform/dev/setup_keys.sh [hunk 1]
```diff
@@ -3,6 +3,9 @@
 # Retrieve keys from your netrc, specified as
 # machine sfo2.digitaloceanspaces.com login KEY password SECRET
 # Source this file before running terraform commands
-netrc_string=$(grep digitalocean.dev.treetracker.org ~/.netrc)
+
+grep_string=$(cat backend.tf | grep endpoint | awk -F'=' '{print $2}' | awk -F'/' '{print $NF}' | sed 's/"//g')
+
+netrc_string=$(grep ${grep_string} ~/.netrc)
```
Comment (dadiorchen) [inline database-grants/terraform/dev/setup_keys.sh:9]: @sanketnadkarni is this relevant to this issue (fix version), if not, please split it into another pull request, thank you!

A good practice of raising pull request: make it small and focused.

**Comment by dadiorchen** [comment]
This pull request looks good for me, accept the comment I made about

File: database-grants/terraform/dev/setup_keys.sh (hunk 1 above)
Comment (sanketnadkarni) [reply database-grants/terraform/dev/setup_keys.sh:9]: this may not be directly related though affects the terraform working. 
i can add this into a subsequent PR though.

**Comment by sanketnadkarni** [comment]
> This pull request looks good for me, accept the comment I made about

Cool thanks for the review @dadiorchen  , I have addressed your comments, if all looks good , I will go ahead with merging this one and create another one for the remaining environments.

**Comment by sanketnadkarni** [comment]
@dadiorchen  could you approve the PR

**Review by dadiorchen — APPROVED** [review]
